storage/shard.go

```diff
@@ -354,27 +361,61 @@ func (s *dbShard) Close() error {
 		stopwatch.Stop()
 	}()
 
+	// NB(prateek): wait till any existing ticks are finished. In the usual
+	// case, no other ticks are running, and tickWg count is at 0, so the
+	// call to Wait() will return immediately.
+	// In the case when there is an existing Tick running, the count for
+	// tickWg will be > 0, and we'll wait until it's reset to zero, which
+	// will happen because earlier in this function we set the shard state
+	// to dbShardStateClosing, which triggers an early termination of
+	// any active ticks.
+	s.tickWg.Wait()
+
 	// NB(r): Asynchronously we purge expired series to ensure pressure on the
 	// GC is not placed all at one time.  If the deadline is too low and still
 	// causes the GC to impact performance when closing shards the deadline
 	// should be increased.
 	cancellable := context.NewNoOpCanncellable()
 	softDeadline := s.opts.TickInterval()
-	s.tickAndExpire(cancellable, softDeadline, tickPolicyForceExpiry)
+	s.tickAndExpire(cancellable, softDeadline, tickPolicyForceExpiry, false)
```

Comment:
Probably want to check if this returns an error and return it yeah?

Reply:
nice catch, will do

Reply:
done